cmd/aro/update_role_sets.go

```diff
@@ -0,0 +1,290 @@
+	var OperatorName OperatorName
+
+	if err := getEnvironmentData(envKey, OperatorName); err != nil {
+		return "", err
+	}
+
+	return OperatorName, nil
+}
+
+func getRoleDefinitionName() (RoleDefinitionName, error) {
+	const envKey = envRoleDefinitionName
+	var RoleDefinitionName RoleDefinitionName
+
+	if err := getEnvironmentData(envKey, RoleDefinitionName); err != nil {
+		return "", err
+	}
+
+	return RoleDefinitionName, nil
+}
+
+func getRoleDefinitionID() (RoleDefinitionID, error) {
+	const envKey = envRoleDefinitionId
+	var RoleDefinitionID RoleDefinitionID
+
+	if err := getEnvironmentData(envKey, RoleDefinitionID); err != nil {
+		return "", err
+	}
+
+	return RoleDefinitionID, nil
+}
+
+func getServiceAccounts() (ServiceAccounts, error) {
+	const envKey = envServiceAccounts
+	var ServiceAccounts ServiceAccounts
+
+	if err := getEnvironmentData(envKey, ServiceAccounts); err != nil {
+		return nil, err
+	}
+
+	return ServiceAccounts, nil
+}
+
+func getRoleSetFromEnv() ([]api.PlatformWorkloadIdentityRoleSet, error) {
+	openShiftVersion, err := getOpenShiftVersion()
+	if err != nil {
+		return []api.PlatformWorkloadIdentityRoleSet{}, err
+	}
+
+	operatorName, err := getOperatorName()
+	if err != nil {
+		return []api.PlatformWorkloadIdentityRoleSet{}, err
+	}
+
+	serviceAccounts, err := getServiceAccounts()
+	if err != nil {
+		return []api.PlatformWorkloadIdentityRoleSet{}, err
+	}
+
+	roleDefinitionName, err := getRoleDefinitionName()
+	if err != nil {
+		return []api.PlatformWorkloadIdentityRoleSet{}, err
+	}
+
+	roleDefinitionId, err := getRoleDefinitionID()
+	if err != nil {
+		return []api.PlatformWorkloadIdentityRoleSet{}, err
+	}
+
+	platformWorkloadIdentityRoleSet := []api.PlatformWorkloadIdentityRoleSet{
+		{
+			Properties: api.PlatformWorkloadIdentityRoleSetProperties{
+				OpenShiftVersion: string(openShiftVersion),
+				PlatformWorkloadIdentityRoles: []api.PlatformWorkloadIdentityRole{
+					{
+						OperatorName:       string(operatorName),
+						ServiceAccounts:    serviceAccounts,
+						RoleDefinitionName: string(roleDefinitionName),
+						RoleDefinitionID:   string(roleDefinitionId),
+					},
+				},
+			},
+		},
+	}
+
+	return platformWorkloadIdentityRoleSet, nil
+}
+
+// 3 - Get the existing role set documents, if existing
+// Mostly copied from update_ocp_versions.go
+func getPlatformWorkloadIdentityRoleSetDatabase(ctx context.Context, log *logrus.Entry) (database.PlatformWorkloadIdentityRoleSets, error) {
+	_env, err := env.NewCore(ctx, log, env.COMPONENT_UPDATE_OCP_VERSIONS)
+	if err != nil {
+		return nil, err
+	}
+
+	msiToken, err := _env.NewMSITokenCredential()
+	if err != nil {
+		return nil, fmt.Errorf("MSI Authorizer failed with: %s", err.Error())
+	}
+
+	msiKVAuthorizer, err := _env.NewMSIAuthorizer(_env.Environment().KeyVaultScope)
+	if err != nil {
+		return nil, fmt.Errorf("MSI KeyVault Authorizer failed with: %s", err.Error())
+	}
+
+	// is this needed in development, or just prod?
```

Comment:
I think the OCP versions script is only used in prod, and we can assume this one will only be used in prod too. For dev, you can just make API calls to set up whatever OCP versions or role sets you need.